iiwa_front:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000  # in Hz

  joint_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    state_publish_rate: 1000

  joint_torque_trajectory_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    gains:
      F_joint_1: { p: 2800.0, i: 80.0, d: 100.0, i_clamp: 320, antiwindup: true }
      F_joint_2: { p: 3500.0, i: 20.0, d: 80.0,  i_clamp: 320, antiwindup: true }
      F_joint_3: { p: 2800.0, i: 50.0, d: 40.0,  i_clamp: 176, antiwindup: true }
      F_joint_4: { p: 1500.0, i: 10.0, d: 25.0,  i_clamp: 176, antiwindup: true }
      F_joint_5: { p: 1800.0, i: 10.0, d: 30.0,  i_clamp: 110, antiwindup: true }
      F_joint_6: { p: 1200.0, i: 15.0, d: 15.0,  i_clamp: 40,  antiwindup: true }
      F_joint_7: { p: 800.0,  i: 15.0, d: 15.0,  i_clamp: 40,  antiwindup: true }
    state_publish_rate: 1000

  gazebo_ros_control:
    pid_gains:
      F_joint_1: { p: 1500, i: 0.0, d: 20.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      F_joint_2: { p: 3000, i: 0.0, d: 50.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      F_joint_3: { p: 1000, i: 0.0, d: 10.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      F_joint_4: { p: 1500, i: 0.0, d: 20.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      F_joint_5: { p: 800,  i: 0.0, d: 3.0,  i_clamp: 110, antiwindup: true } # Parameter for simulation
      F_joint_6: { p: 500,  i: 0.0, d: 5.0,  i_clamp: 40,  antiwindup: true } # Parameter for simulation
      F_joint_7: { p: 500,  i: 0.0, d: 0.5,  i_clamp: 40,  antiwindup: true } # Parameter for simulation

  joint_feedforward_trajectory_controller:
    type: feedforward_controllers/JointTrajectoryController
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    gains:
      F_joint_1: { p: 1500, i: 0.0, d: 20.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      F_joint_2: { p: 3000, i: 0.0, d: 50.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      F_joint_3: { p: 1000, i: 0.0, d: 10.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      F_joint_4: { p: 1500, i: 0.0, d: 20.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      F_joint_5: { p: 800,  i: 0.0, d: 3.0,  i_clamp: 110, antiwindup: true } # Parameter for simulation
      F_joint_6: { p: 500,  i: 0.0, d: 5.0,  i_clamp: 40,  antiwindup: true } # Parameter for simulation
      F_joint_7: { p: 500,  i: 0.0, d: 0.5,  i_clamp: 40,  antiwindup: true } # Parameter for simulation
    state_publish_rate: 1000

  adrc_trajectory_controller:
    type: adrc_controllers/JointTrajectoryController
    state_publish_rate: 1000
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    parameters:
      F_joint_1: { b: 9.0,  omega_c: 70.,  omega_o: 300, k: 800, u_max: 160}
      F_joint_2: { b: 3.0,  omega_c: 50.,  omega_o: 200, k: 300, u_max: 160}
      F_joint_3: { b: 5.0,  omega_c: 100., omega_o: 400, k: 600, u_max: 90}
      F_joint_4: { b: 3.0,  omega_c: 50.,  omega_o: 200, k: 200, u_max: 90}
      F_joint_5: { b: 0.5,  omega_c: 100., omega_o: 400, k: 400, u_max: 60}
      F_joint_6: { b: 0.4,  omega_c: 50.,  omega_o: 200, k: 100, u_max: 20}
      F_joint_7: { b: 0.25, omega_c: 100., omega_o: 400, k: 300, u_max: 20}


iiwa_back:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000  # in Hz

  joint_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    state_publish_rate: 1000

  joint_torque_trajectory_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    gains:
      B_joint_1: { p: 2800.0, i: 80.0, d: 100.0, i_clamp: 320, antiwindup: true }
      B_joint_2: { p: 3500.0, i: 20.0, d: 80.0,  i_clamp: 320, antiwindup: true }
      B_joint_3: { p: 2800.0, i: 50.0, d: 40.0,  i_clamp: 176, antiwindup: true }
      B_joint_4: { p: 1500.0, i: 10.0, d: 25.0,  i_clamp: 176, antiwindup: true }
      B_joint_5: { p: 1800.0, i: 10.0, d: 30.0,  i_clamp: 110, antiwindup: true }
      B_joint_6: { p: 1200.0, i: 15.0, d: 15.0,  i_clamp: 40,  antiwindup: true }
      B_joint_7: { p: 800.0,  i: 15.0, d: 15.0,  i_clamp: 40,  antiwindup: true }
    state_publish_rate: 1000

  gazebo_ros_control:
    pid_gains:
      B_joint_1: { p: 1500, i: 0.0, d: 20.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      B_joint_2: { p: 3000, i: 0.0, d: 50.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      B_joint_3: { p: 1000, i: 0.0, d: 10.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      B_joint_4: { p: 1500, i: 0.0, d: 20.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      B_joint_5: { p: 800,  i: 0.0, d: 3.0,  i_clamp: 110, antiwindup: true } # Parameter for simulation
      B_joint_6: { p: 500,  i: 0.0, d: 5.0,  i_clamp: 40,  antiwindup: true } # Parameter for simulation
      B_joint_7: { p: 500,  i: 0.0, d: 0.5,  i_clamp: 40,  antiwindup: true } # Parameter for simulation

  joint_feedforward_trajectory_controller:
    type: feedforward_controllers/JointTrajectoryController
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    gains:
      B_joint_1: { p: 1500, i: 0.0, d: 20.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      B_joint_2: { p: 3000, i: 0.0, d: 50.0,   i_clamp: 320, antiwindup: true } # Parameter for simulation
      B_joint_3: { p: 1000, i: 0.0, d: 10.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      B_joint_4: { p: 1500, i: 0.0, d: 20.0, i_clamp: 176, antiwindup: true } # Parameter for simulation
      B_joint_5: { p: 800,  i: 0.0, d: 3.0,  i_clamp: 110, antiwindup: true } # Parameter for simulation
      B_joint_6: { p: 500,  i: 0.0, d: 5.0,  i_clamp: 40,  antiwindup: true } # Parameter for simulation
      B_joint_7: { p: 500,  i: 0.0, d: 0.5,  i_clamp: 40,  antiwindup: true } # Parameter for simulation
    state_publish_rate: 1000

  adrc_trajectory_controller:
    type: adrc_controllers/JointTrajectoryController
    state_publish_rate: 1000
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    parameters:
      B_joint_1: { b: 8.0,  omega_c: 100., omega_o: 500, k: 1000, u_max: 160}
      B_joint_2: { b: 3.0,  omega_c: 50.,  omega_o: 200, k: 300, u_max: 160}
      B_joint_3: { b: 5.0,  omega_c: 100., omega_o: 400, k: 600, u_max: 90}
      B_joint_4: { b: 3.0,  omega_c: 50.,  omega_o: 200, k: 200, u_max: 90}
      B_joint_5: { b: 0.5,  omega_c: 100., omega_o: 400, k: 400, u_max: 60}
      B_joint_6: { b: 0.4,  omega_c: 50.,  omega_o: 200, k: 100, u_max: 20}
      B_joint_7: { b: 0.25, omega_c: 100., omega_o: 400, k: 300, u_max: 20}